### DISCUSSION TITLE
please add prop "resetAfterSubmit" to Form element

### DISCUSSION THREAD

**Discussion by xaota** [remix-run/react-router#12540] [Proposals]
please add prop "resetAfterSubmit" to Form element

**Comment by gustavopch** [comment]
I think adding more details on why you need this, how exactly it should work, what you're doing currently as a workaround, etc. will increase the chance of getting attention from the team.

**Reply by xaota** [reply]
I want reset form (clear all fields) after successful submit, but how i can know submit status?

- if action was throw -> route drop to ErrorBoundary
- if action has redirect, you can't return actionData and can't use It to clear form
